src/Resolver/Proxy.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Upward\Resolver;
+
+use Magento\Upward\Definition;
+use Zend\Http\Client;
+
+class Proxy extends AbstractResolver
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getIndicator(): string
+    {
+        return 'proxy';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isValid(Definition $definition): bool
+    {
+        $rootDefinition = $this->getIterator()->getRootDefinition();
```

Comment:
Why are you looking at the root definition and validating a key from there? Should you be looking at the definition passed and checking it's required properties?

Reply:
My mistake, I think you're correct. In looking at the Venia configs, I mistook `use: proxy` in conditional resolvers as a static root level config all proxies shared. Paired with my misunderstanding of the indicator too, this makes a lot more sense now, good catch.